### ISSUE TITLE
Is it recommended or good idea to have 2 diff./same circuit breaker per message/api call?

### ISSUE THREAD

**Issue by ackjui** [resilience4j/resilience4j#1060]
Resilience4j version:0.13.2

Java version:1.8

Thanks for raising a Resilience4j issue.
Use Case: In my micro-service, circuit breaker is present at a layer where external api call happens and my recorded exceptions are those which can happen during this external api call.
But my service timeout is configured at a layer way above this(where circuit breaker is present/configured) layer. I can't move circuit breaker up to the layer where timeout is configured or vice-versa.

Basically, I want to record this timeout exception happening at different layer.

Is it recommended or good idea to have 2 diff./same circuit breaker per message/api call?

**Comment by RobWin** [comment]
Yes, you can do that. 
But you have to think how exceptions are progapated through two CircuitBreakers.

**Comment by rorawatibs** [comment]
These are the configurations in my case:

`circuitBreakers:
  - name: search
    automaticTransitionFromOpenToHalfOpenEnabled: true
    failureRateThreshold: 80
    waitDurationInOpenState: 30s
    forceDisabled : false
    recordExceptions:
    - com.sun.xml.internal.ws.client.ClientTransportException
    - java.net.SocketTimeoutException
    - java.net.UnknownHostException
    - com.google.common.util.concurrent.UncheckedExecutionException
  - name: searchTimeout
    automaticTransitionFromOpenToHalfOpenEnabled: true
    failureRateThreshold: 80
    waitDurationInOpenState: 30s
    forceDisabled : false
    recordExceptions:
    - java.util.concurrent.TimeoutException`

`search` CB is present down the layer(near external api call), whereas `searchTimeout` CB is present way up the layer(near resource layer).

Since we've clearly defined the `recordedExceptions`, there should be no side effects.

**Comment by RobWin** [comment]
Yes, seems to be ok.
